Prototope/ShapeLayer.swift

```diff
@@ -0,0 +1,424 @@
+	public var lineCapStyle: LineCapStyle = .Butt {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	/** Represents the types of join styles path segments will show at their joins. */
+	public enum LineJoinStyle {
+		
+		/** Lines will be joined with a miter style. */
+		case Miter
+		
+		/** Lines will be joined with a round style. */
+		case Round
+		
+		/** Line joins will have beveled edges. */
+		case Bevel
+		
+		internal func joinStyleString() -> String {
+			switch self {
+			case Miter:
+				return kCALineJoinMiter
+			case Round:
+				return kCALineJoinRound
+			case Bevel:
+				return kCALineJoinBevel
+			}
+		}
+	}
+	
+	
+	/** The line join style for path lines. Defaults to LineJoinStyle.Miter. */
+	public var lineJoinStyle: LineJoinStyle = .Miter {
+		didSet {
+			shapeViewLayerStyleChanged()
+		}
+	}
+	
+	
+	// MARK: - Private details
+	
+	private var shapeViewLayer: CAShapeLayer {
+		return self.view.layer as! CAShapeLayer
+	}
+	
+	
+	private class ShapeView: UIView {
+		override class func layerClass() -> AnyClass {
+			return CAShapeLayer.self
+		}
+	}
+	
+	
+	private func shapeViewLayerStyleChanged() {
+		var layer = self.shapeViewLayer
+		layer.lineCap = self.lineCapStyle.capStyleString()
+		layer.lineJoin = self.lineJoinStyle.joinStyleString()
+		
+		if let fillColor = fillColor {
+			self.shapeViewLayer.fillColor = fillColor.CGColor
+		} else {
+			self.shapeViewLayer.fillColor = nil
+		}
+		
+		
+		if let strokeColor = strokeColor {
+			self.shapeViewLayer.strokeColor = strokeColor.CGColor
+		} else {
+			self.shapeViewLayer.strokeColor = nil
+		}
+		
+		self.shapeViewLayer.lineWidth = CGFloat(strokeWidth)
+	}
+	
+	
+	private var bezierPath: UIBezierPath {
+		
+		let bezierPath = UIBezierPath()
+		var isFirstSegment = true
+		var currentPoint = Point()
+		var previousPoint = Point()
+		var currentHandleIn = Point()
+		var currentHandleOut = Point()
+		
+		func drawSegment(segment: Segment) {
+			currentPoint = segment.point
+			
+			if isFirstSegment {
+				bezierPath.moveToPoint(CGPoint(currentPoint))
+				isFirstSegment = false
+			} else {
+				if let segmentHandleIn = segment.handleIn {
+					currentHandleIn = currentPoint + segmentHandleIn
+				} else {
+					currentHandleIn = currentPoint
+				}
+				
+				
+				if currentHandleIn == currentPoint && currentHandleOut == previousPoint {
+					bezierPath.addLineToPoint(CGPoint(currentPoint))
+				} else {
+					bezierPath.addCurveToPoint(CGPoint(currentPoint), controlPoint1: CGPoint(currentHandleOut), controlPoint2: CGPoint(currentHandleIn))
+				}
+			}
+			
+			previousPoint = currentPoint
+			if let segmentHandleOut = segment.handleOut {
+				currentHandleOut = previousPoint + segmentHandleOut
+			} else {
+				currentHandleOut = previousPoint
+			}
+			
+		}
+		for segment in self.segments {
+			drawSegment(segment)
+		}
+		
+		if self.closed && self.segments.count > 0 {
+			drawSegment(self.segments[0])
+			bezierPath.closePath()
```

Comment:
Isn't this unnecessary given that you've just drawn a segment back to the first segment?